setup_posix.py

```diff
@@ -9,9 +9,7 @@
 # of mysql_config
 
 def dequote(s):
-    if s[0] in "\"'" and s[0] == s[-1]:
-        s = s[1:-1]
-    return s
+    return s.strip("'\"")
```

Comment:
I don't like this code.
Raising exception for wrong input is good thing.  It's not reason for breaking our code.

Reply:
This could change so that line 12 is `if s and s[0] in "\"'" and s[0] == s[-1]:` .  While raising exceptions is the way to go: 

1.  `IndexError` is not informative about the root cause -- so this could be caught higher up and rethrown with more context if it is to be preserved.
2.  Removing any possibility of the error (as in my proposed change(s)) allowed me to proceed to install the package, so the error may be too conservative.

Reply:
First of all, this issue is very special.  Once MySQL fixed, it never happen.  So IndexError is OK.

And "Errors should never pass silently."

You should fix `mysql_config` side, not here.